node/realgen/tString.go

```diff
@@ -0,0 +1,229 @@
+package realgen
+
+import (
+	ipld "github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/node/mixins"
+	"github.com/ipld/go-ipld-prime/schema"
+)
+
+// Code generated go-ipld-prime DO NOT EDIT.
+
+type _String struct{ x string }
+type String = *_String
+
+func (n String) String() string {
+	return n.x
+}
+func (_String__Style) fromString(w *_String, v string) error {
+	*w = _String{v}
+	return nil
+}
+func (_String__Style) FromString(v string) (String, error) {
+	n := _String{v}
+	return &n, nil
+}
+
+type _String__Maybe struct {
+	m schema.Maybe
+	v String
+}
+type MaybeString = *_String__Maybe
+
+func (m MaybeString) IsNull() bool {
+	return m.m == schema.Maybe_Null
+}
+func (m MaybeString) IsUndefined() bool {
+	return m.m == schema.Maybe_Absent
+}
+func (m MaybeString) Exists() bool {
+	return m.m == schema.Maybe_Value
+}
+func (m MaybeString) AsNode() ipld.Node {
+	switch m.m {
+	case schema.Maybe_Absent:
+		return ipld.Undef
+	case schema.Maybe_Null:
+		return ipld.Null
+	case schema.Maybe_Value:
+		return m.v
+	default:
+		panic("unreachable")
+	}
+}
+func (m MaybeString) Must() String {
+	if !m.Exists() {
+		panic("unbox of a maybe rejected")
+	}
+	return m.v
+}
+
+var _ ipld.Node = (String)(&_String{})
+var _ schema.TypedNode = (String)(&_String{})
+
+func (String) ReprKind() ipld.ReprKind {
+	return ipld.ReprKind_String
+}
+func (String) LookupString(string) (ipld.Node, error) {
+	return mixins.String{"realgen.String"}.LookupString("")
+}
+func (String) Lookup(ipld.Node) (ipld.Node, error) {
+	return mixins.String{"realgen.String"}.Lookup(nil)
+}
+func (String) LookupIndex(idx int) (ipld.Node, error) {
+	return mixins.String{"realgen.String"}.LookupIndex(0)
+}
+func (String) LookupSegment(seg ipld.PathSegment) (ipld.Node, error) {
+	return mixins.String{"realgen.String"}.LookupSegment(seg)
+}
+func (String) MapIterator() ipld.MapIterator {
+	return nil
+}
+func (String) ListIterator() ipld.ListIterator {
+	return nil
+}
+func (String) Length() int {
+	return -1
+}
+func (String) IsUndefined() bool {
+	return false
+}
+func (String) IsNull() bool {
+	return false
+}
+func (String) AsBool() (bool, error) {
+	return mixins.String{"realgen.String"}.AsBool()
+}
+func (String) AsInt() (int, error) {
+	return mixins.String{"realgen.String"}.AsInt()
+}
+func (String) AsFloat() (float64, error) {
+	return mixins.String{"realgen.String"}.AsFloat()
+}
+func (n String) AsString() (string, error) {
```

Comment:
thoughts on also adding the [`String()`](https://golang.org/src/fmt/print.go#L62) method so values are auto-converted by fmt?

Reply:
Uncertain!  What do you think?

There's also the possibility of types having a `String() string` method (no error return, like `AsString` has) when their kind is in fact `string`.  But I wonder if that should be reserved as an interface-castable indictation of string kinds?

Perhaps considering that an indicative interface is pretty foolish considering `fmt.Stringer` already has the contract it does, though.

Reply:
There's also the question of whether `String() string` should exist on things such a struct type with stringjoin representation.  On the representation node, the answer is definitely 'yes'.  On the type-level node, I guess it would be {insert answer to whether we just always emit a fmt.Stringer}.

Reply:
I think, as long as there's a story, be it implementing `Formatter`, or one of the other interfaces that allow the various common debugging methods to display the value in a sane way I don't have strong attachments to `Stringer` in particular.

I see your point about helping programmers do good defensive programming by forcing them to consider the error, and at the same time see why that's often not ergonomic in Go. I would probably provide common implementer of interfaces to allow that final value to be directly passed into recipient functions with one less extraction of error.